updates/updates.py

```diff
@@ -0,0 +1,112 @@
+from datetime import datetime, timezone, timedelta
+from shared.shared import logging
+import os
+import asyncio
+import json
+import requests
+
+def load_posted_events():
+    try:
+        file_path = os.path.join(os.path.dirname(__file__), "..", "updates", "posted_events.json")
+        logging.info(f"Loading events from: {file_path}")
+        with open(file_path, "r") as file:
+            return json.load(file)
+    except FileNotFoundError:
+        return []
+
+def save_posted_events(posted_events):
+    try:
+        file_path = os.path.join(os.path.dirname(__file__), "..", "updates", "posted_events.json")
+
+        logging.info(f"Saving events to: {file_path}")
+
+        with open(file_path, "w") as file:
+            json.dump(posted_events, file)
+            
+    except Exception as e:
+        logging.error(f"Error saving posted events: {e}")
+    
+def format_event(event):
+    # Format the event start time for Discord time
+    guild_id = int(os.getenv("GUILD_ID"))
+    event_url = f"https://discord.com/events/{guild_id}/{event.id}"
+
+    formatted_event = (
+        f"\n"
+        f" **{event.name}**\n"
+        f"\n"
+        f"To request someones attendance, react to this message with their emoji! \n"
+        f"\n"
+        f":link:** Event Link** {event_url} :link:\n"
+    )
+    return formatted_event
+
+async def notify_new_event(bot, event):
+    guild_id = int(os.getenv("GUILD_ID"))
+    guild = bot.get_guild(guild_id)
+
+    if guild:
+        # Wait for 30 mins before sending the notification
+        await asyncio.sleep(60 * 30)
+
+        # Fetch the event again to get the updated details
+        event = await guild.fetch_scheduled_event(event.id)
+        formatted_event = format_event(event)
+
+        channel_id = int(os.getenv("GENERAL_CHANNEL_ID"))
+        channel = guild.get_channel(channel_id)
+
+        if channel:
+            # Send the notification and capture the Message object
+            await channel.send(f"🌺 **__Newly Created Event__** 🌺 \n{formatted_event}")
+
+        else:
+            logging.info(f"Event channel not found")
+    else:
+        logging.info(f"Guild not found")
+
+async def check_upcoming_events(guild, time_range=None):
+    current_time = datetime.now().astimezone(timezone.utc)
+    events = guild.scheduled_events
+    upcoming_events = []
+
+    # Check if the event's start time is in the future (after or equal to the current time)
+    # And if a time range is specified, ensure the time difference is within that range
+    for event in events:
+        if time_range:
+            # Fetch the event again to get the updated user_count
+            event = await guild.fetch_scheduled_event(event.id)
+
+            time_difference = event.start_time - current_time
+            if 0 <= time_difference.total_seconds() <= time_range:
+                upcoming_events.append(event)
+        else:
+            upcoming_events.append(event)
```

Comment:
Pls explain this section 
You are appending your event in any case

Reply:
> Pls explain this section You are appending your event in any case

I was thinking if we were going to implement something else related to the events we may want to return them anyway. But I will rework this logic per my other comment about the purpose / logic of the function